tests/api/v2/handlers/test_abilities_api.py

```diff
@@ -0,0 +1,142 @@
+import pytest
+
+from http import HTTPStatus
+
+from app.objects.c_ability import Ability
+from app.objects.secondclass.c_executor import Executor, ExecutorSchema
+from app.utility.base_service import BaseService
+
+
+@pytest.fixture
+def test_ability(loop, api_v2_client, executor):
+    executor_linux = executor(name='sh', platform='linux')
+    ability = Ability(ability_id='123', name='Test Ability', executors=[executor_linux],
+                      technique_name='collection', technique_id='1')
+    loop.run_until_complete(BaseService.get_service('data_svc').store(ability))
+    return ability
+
+
+class TestAbilitiesApi:
+    async def test_get_abilities(self, api_v2_client, api_cookies, test_ability):
+        resp = await api_v2_client.get('/api/v2/abilities', cookies=api_cookies)
+        abilities_list = await resp.json()
+        assert len(abilities_list) == 1
+        ability_dict = abilities_list[0]
+        assert ability_dict['ability_id'] == test_ability.ability_id
+        assert ability_dict['name'] == test_ability.name
+        assert ability_dict['technique_name'] == test_ability.technique_name
+        assert len(ability_dict['executors']) == 1
+
+    async def test_unauthorized_get_abilities(self, api_v2_client, test_ability):
+        resp = await api_v2_client.get('/api/v2/abilities')
+        assert resp.status == HTTPStatus.UNAUTHORIZED
+
+    async def test_get_ability_by_id(self, api_v2_client, api_cookies, test_ability):
+        resp = await api_v2_client.get('/api/v2/abilities/123', cookies=api_cookies)
+        ability_dict = await resp.json()
+        assert ability_dict['ability_id'] == test_ability.ability_id
+        assert ability_dict['name'] == test_ability.name
+        assert ability_dict['technique_name'] == test_ability.technique_name
+        assert len(ability_dict['executors']) == 1
+
+    async def test_unauthorized_get_ability_by_id(self, api_v2_client, test_ability):
+        resp = await api_v2_client.get('/api/v2/abilities/123')
+        assert resp.status == HTTPStatus.UNAUTHORIZED
+
+    async def test_get_nonexistent_ability_by_id(self, api_v2_client, api_cookies):
+        resp = await api_v2_client.get('/api/v2/abilities/999', cookies=api_cookies)
+        assert resp.status == HTTPStatus.NOT_FOUND
+
+    async def test_create_ability(self, api_v2_client, api_cookies, mocker, async_return):
+        test_executor_linux = Executor(name='sh', platform='linux', command='whoami')
+        payload = dict(name='new test ability', ability_id='456', tactic='collection', technique_name='collection',
+                       technique_id='1', executors=[ExecutorSchema().dump(test_executor_linux)])
+        resp = await api_v2_client.post('/api/v2/abilities', cookies=api_cookies, json=payload)
+        assert resp.status == HTTPStatus.OK
+        ability_data = await resp.json()
+        assert ability_data.get('name') == payload['name']
```

Comment:
It'll be better to build a dictionary representing the "expected ability" (basically the `payload` dict plus all the default values) and then just assert equality between that and the `ability_data`